_usage/hint_mapping.md

```diff
@@ -0,0 +1,295 @@
+            JOIN t2 ON t1.a1 = t2.a2
+            JOIN t3 ON t1.a1 = t3.a3 OPTION(FORCE ORDER)
+```
+
+#### MAXDOP Hint
+
+You can use a [MAXDOP hint](https://docs.microsoft.com/en-us/sql/t-sql/queries/hints-transact-sql-join?view=sql-server-ver16) as a query hint specified with the OPTION clause. To apply this hint, you must first set the PostgreSQL `max_parallel_workers_per_gather` GUC to the value specified in the hint; for example:
+
+```sql
+SELECT * FROM t1 WHERE a1=1 OPTION(MAXDOP 4)
+```
+
+### Example
+
+First, create the sample environment by creating test tables and turning on the `BABELFISH_SHOWPLAN_ALL` GUC:
+
+```sql
+1> CREATE TABLE t1 (a1 INT PRIMARY KEY, b1 INT);
+2> CREATE TABLE t2 (a2 INT PRIMARY KEY, b2 INT);
+3> SELECT set_config('babelfishpg_tsql.enable_hint_mapping', 'on', false);
+4> GO
+
+1> SET BABELFISH_SHOWPLAN_ALL ON;
+2> GO
+```
+
+Then, query the tables with a simple join, with no hints:
+
+```sql
+1> SELECT * FROM t1 JOIN t2 ON t1.a1 = t2.a2; -- NO HINTS
+2> GO
+QUERY PLAN
+------------------------------------------------------------------
+Query Text: SELECT * FROM t1 JOIN t2 ON t1.a1 = t2.a2
+Hash Join  (cost=60.85..99.39 rows=2260 width=16)
+  Hash Cond: (t1.a1 = t2.a2)
+  ->  Seq Scan on t1  (cost=0.00..32.60 rows=2260 width=8)
+  ->  Hash  (cost=32.60..32.60 rows=2260 width=8)
+        ->  Seq Scan on t2  (cost=0.00..32.60 rows=2260 width=8)
+```
+
+Then, query the tables using a merge join; the result set shows the execution costs:
+
+```sql
+1> SELECT * FROM t1 INNER MERGE JOIN t2 ON t1.a1 = t2.a2; -- USE MERGE JOIN
+2> GO
+QUERY PLAN
+---------------------------------------------------------------------------------------------------
+Query Text: /*+ MergeJoin(t1 t2) Leading(t1 t2)*/ SELECT * FROM t1 INNER JOIN t2 ON t1.a1 = t2.a2
+Merge Join  (cost=0.31..190.01 rows=2260 width=16)
+  Merge Cond: (t1.a1 = t2.a2)
+  ->  Index Scan using t1_pkey on t1  (cost=0.15..78.06 rows=2260 width=8)
+  ->  Index Scan using t2_pkey on t2  (cost=0.15..78.06 rows=2260 width=8)        
+        
+```
+
+Then, query the same tables using a loop join:
+
+```sql
+1> SELECT * FROM t1 JOIN t2 ON t1.a1 = t2.a2 OPTION(LOOP JOIN); -- USE LOOP JOIN
+2> GO
+QUERY PLAN
+----------------------------------------------------------------------------------------------------------------
+Query Text: /*+ Set(enable_hashjoin off) Set(enable_mergejoin off) */ SELECT * FROM t1 JOIN t2 ON t1.a1 = t2.a2
+Nested Loop  (cost=0.15..466.41 rows=2260 width=16)
+  ->  Seq Scan on t1  (cost=0.00..32.60 rows=2260 width=8)
+  ->  Index Scan using t2_pkey on t2  (cost=0.15..0.19 rows=1 width=8)
+        Index Cond: (a2 = t1.a1)
+
+1> SET BABELFISH_SHOWPLAN_ALL OFF;
+2> GO
+```
+
+
+### Limitations
+
+`pg_hint_plan` reads hints from only the first block comment, so if the query has any existing hints as a `pg_hint_plan` comment, it will be ignored. You can disable the feature using the GUC if the existing `pg_hint_plan` comment block should be considered:
```

Comment:
1. If a query has both T-SQL query hints (only the supported ones) and an existing pg-hint-plan comment style hint, only one of them will be applied.
- If the GUC is on, T-SQL query hints will be applied
- If the GUC is off, existing pg-hint-plan comment style hint will be applied

2. However, if the query only has pg-hint-plan comment style hint, it will be applied with the GUC turned on/off